waspc/tools/bash-completion

```diff
@@ -0,0 +1,27 @@
+_wasp()
```

Comment:
Btw I found this (from here https://metacpan.org/pod/Complete::Bash):

> And yet another source is an external command (-C) including, from a Perl script. The command receives two environment variables: COMP_LINE (string, raw command-line) and COMP_POINT (integer, cursor location). Program must split COMP_LINE into words, find the word to be completed, complete that, and return the list of words one per-line to STDOUT. An example:

So maybe it makes sense to not have bash script for this, but instead have this logic directly in our Haskell CLI code?
The only thing that remains then is to run the `complete ...` command at the right place/time (which I am not yet sure about what it is?).

Reply:
I manage to move all logics that stayed in `tools/bash-completion` into the Haskell cli with some tweaks. When the CLI sees `wasp <command>`, it will call `wasp commands all` to generate all options for the first argument. With the case of `wasp db <command>`, it will simply fetch the keyword `db` i.e `wasp commands db` to looks for options.

Reply:
My idea in the last review was to actually move all the logic into Haskell, so that only `complete -o default -o nospace -C "wasp command"` would be present in the bash-completion file, nothing else.

Haskell code would check the env vars COMP_WORDS and COMP_CWORD and would return results of auto-completion to the STDOUT, one per line.

This way, we would never have to update the bash-completion file, all future updates would be happening in the Haskell code - for example if we decide to add one more level of commands, so we have `wasp db dev <command>`, this is something we would not have to modify the bash script for in order to support it.

The solution you implemented is half way there - bash script assumes that wasp has top-lvl commands or subcommands one lvl deep, and as long as that does not change, we don't have to modify this bash script.
This is an assumption that is fairly decent, as it is not very likely we will be adding deeper levels, but stilll, it might happen!

I wonder, are there any bad sides to trying to put all this logic into the Haskell CLI? Bloating the CLI?

If you think putting it all in Haskell would be too much, I understand, in that case we can leave it as it right now and go with it!

Reply:
Unfortunately, `COMP_WORDS` and `COMP_CWORD` are not env vars, they are some magic from bash completion functionalities. We can use the similar approach from  https://metacpan.org/pod/Complete::Bash: 

> And yet another source is an external command (-C) including, from a Perl script. The command receives two environment variables: COMP_LINE (string, raw command-line) and COMP_POINT (integer, cursor location). Program must split COMP_LINE into words, find the word to be completed, complete that, and return the list of words one per-line to STDOUT

but it looks like it will be too much on the Haskell side so I think we can leave it there for now. Its flexibility is good enough to cover most cases.

Reply:
Sorry, I made a mistake there, we would have to use COMP_LINE instead of COMP_WORDS and COMP_CWORD, yes!

Ok, makes sense, let's not overcomplicate it for now and go with this as a first version.

One thing that remains and which I am not sure about still is how will users install bash completion for Wasp. I asked more about it here: https://github.com/wasp-lang/wasp/pull/224#pullrequestreview-658223017 .